force-app/main/selector/HouseholdSelector.cls

```diff
@@ -36,7 +36,7 @@ public inherited sharing class HouseholdSelector {
 
     private List<Account> getHouseholdAccountsWithContactsAndSingleOpp(Set<Id> accountIds) {
         List<Account> accounts = [
-                SELECT Id, Type, npe01__SYSTEM_AccountType__c, npe01__One2OneContact__c, (
+                SELECT Id, Type, npe01__SYSTEM_AccountType__c, npe01__One2OneContact__c, RecordTypeId, (
```

Comment:
To help with later filtering, you could also add `RecordType.DeveloperName`. This way the comparison is `RecordType.DeveloperName == :CON_Constants.HH_ACCOUNT_RT_DEVELOPER_NAME`. That avoids having to call other code that finds the recordtype Id for HH_ACCOUNT_RT_DEVELOPER_NAME, unless that RT.Id is already in a custom setting or something.

Reply:
It looks like we're relying on `npe01__SYSTEM_AccountType__c` instead of RecordTypeId here, which should be accurate assuming our code has been filling that field in.